testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/keycloak/saml/adapter/BasicKeycloakOperatorApplication.java

```diff
@@ -0,0 +1,389 @@
+
+		// Self-signed certificate for SAML requests decrypt and SAML assertions signing
+		final SimpleCommandLineBasedKeystoreGenerator.CertificateInfo samlClientCertificate = SimpleCommandLineBasedKeystoreGenerator
+				.generateCertificate(
+						OpenShifts.master().generateHostname(WildflyWithKeycloakSamlAdapterApplication.APP_NAME),
+						WildflyWithKeycloakSamlAdapterApplication.SSO_SAML_CERTIFICATE_NAME,
+						WildflyWithKeycloakSamlAdapterApplication.SSO_SAML_KEYSTORE_PASSWORD,
+						WildflyWithKeycloakSamlAdapterApplication.SSO_SAML_KEYSTORE_PASSWORD,
+						Collections.emptyList());
+
+		// Secret detailed structure can be checked in: https://www.keycloak.org/operator/advanced-configuration#_truststores
+		// E.g.:
+		// apiVersion: v1
+		// kind: Secret
+		// metadata:
+		//   name: my-secret
+		// stringData:
+		//   cert.pem: |
+		//     -----BEGIN CERTIFICATE-----
+		//     ...
+		Secret truststoreSecret = new SecretBuilder()
+				.withNewMetadata()
+				.withName(APP_NAME + "-truststore")
+				.withLabels(Collections.singletonMap("app", APP_NAME))
+				.endMetadata()
+				.addToData(Map.of("cert.pem",
+						Base64.getEncoder()
+								.encodeToString(FileUtils.readFileToByteArray(samlClientCertificate.certificate.toFile()))))
+				.build();
+		secrets.add(truststoreSecret);
+
+		// Private Key + Self-signed certificate to encrypt traffic to the Keycloak service
+		// https://www.keycloak.org/docs/latest/server_admin/index.html#loading-keys-from-a-java-keystore
+		final SimpleCommandLineBasedKeystoreGenerator.CertificateInfo httpsKeystore = SimpleCommandLineBasedKeystoreGenerator
+				.generateCertificate(
+						OpenShifts.master().generateHostname(APP_NAME),
+						HTTPS_CERTIFICATE_NAME,
+						HTTPS_KEYSTORE_PASSWORD,
+						HTTPS_KEYSTORE_PASSWORD,
+						Collections.emptyList());
+
+		Secret httpsSecret = new SecretBuilder()
+				.withNewMetadata()
+				.withName(APP_NAME + "-keystore")
+				.withLabels(Collections.singletonMap("app", APP_NAME))
+				.endMetadata()
+				.addToData(Map.of("tls.crt",
+						Base64.getEncoder().encodeToString(FileUtils.readFileToByteArray(httpsKeystore.certificate.toFile()))))
+				.addToData(Map.of("tls.key",
+						Base64.getEncoder().encodeToString(FileUtils.readFileToByteArray(httpsKeystore.privatekey.toFile()))))
+				.build();
+		secrets.add(httpsSecret);
+
+		final String hostName = OpenShifts.master().generateHostname(APP_NAME);
+		keycloak = new KeycloakBuilder()
+				.withNewMetadata()
+				.withName(APP_NAME)
+				.withLabels(Collections.singletonMap("app", APP_NAME))
+				.endMetadata()
+				.withNewSpec()
+				.withInstances(KEYCLOAK_INSTANCES)
+				// We don't check hostnames in test certificates (usually hostnames are not yet available when they are created)
+				// See:
+				// - https://www.keycloak.org/server/keycloak-truststore#_hostname_verification_policy
+				// - https://www.keycloak.org/operator/advanced-configuration#_additional_options
+				.withAdditionalOptions(
```

Comment:
Is this needed at all? I mean, which host names should Keycloak verify? Maybe I am missing something?